### PR TITLE
Make MiniChefV21 implement IMiniChefV2

### PR DESCRIPTION
**Pull request by bleaker0x**
## Description

Modify IMiniChefV2 enough that MiniChefV21 can implement it

## Checklist

- [x] New Contracts have been tested
- [x] Lint has been run
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: contracts/bridge/MiniChefV21.sol [hunk 1]
```diff
@@ -26,32 +29,32 @@ contract MiniChefV21 is BoringOwnable, BoringBatchable {
     /// @notice Info of each MCV2 user.
     /// `amount` LP token amount the user has provided.
     /// `rewardDebt` The amount of SYNAPSE entitled to the user.
-    struct UserInfo {
-        uint256 amount;
-        int256 rewardDebt;
-    }
+    //    struct UserInfo {
```
Comment (trajan0x) [inline contracts/bridge/MiniChefV21.sol:32]: Lets ditch thesse and move these comments to the interface

File: contracts/bridge/interfaces/IMiniChefV2.sol [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {IRewarder} from "./IRewarder.sol";
 interface IMiniChefV2 {
     struct UserInfo {
         uint256 amount;
-        uint256 rewardDebt;
+        int256 rewardDebt;
```
Comment (trajan0x) [inline contracts/bridge/interfaces/IMiniChefV2.sol:11]: Issue here is now this no longer matches the canonical minichef2 (sushi + https://github.com/synapsecns/synapse-contracts/blob/master/contracts/bridge/MiniChefV2.sol)

**Review by trajan0x — CHANGES_REQUESTED** [review]

File: contracts/bridge/interfaces/IMiniChefV2.sol (hunk 2 above)
Comment (bleaker0x) [reply contracts/bridge/interfaces/IMiniChefV2.sol:11]: From that link:

MiniChefV2: 

```
    struct UserInfo {
        uint256 amount;
        int256 rewardDebt;
    }
```

MiniChefV21.sol/IMiniChefV2:

```
    struct UserInfo {
        uint256 amount;
        int256 rewardDebt;
    }
```

Comment (bleaker0x) [reply contracts/bridge/interfaces/IMiniChefV2.sol:11]: Canonical Sushi (https://github.com/sushiswap/sushiswap/blob/master/contracts/MiniChefV2.sol)

```
    struct UserInfo {
        uint256 amount;
        int256 rewardDebt;
    }
```

File: contracts/bridge/MiniChefV21.sol (hunk 1 above)
Comment (bleaker0x) [reply contracts/bridge/MiniChefV21.sol]: [3b02b24](https://github.com/synapsecns/synapse-contracts/pull/90/commits/3b02b24b51c469114c98d6165e79ceec9fd940b3)

File: contracts/bridge/interfaces/IMiniChefV2.sol (hunk 2 above)
Comment (bleaker0x) [reply contracts/bridge/interfaces/IMiniChefV2.sol:14]: For whatever reason, the original struct in the interface had the non-canonical signature.

**Comment by trajan0x** [comment]
Ready to merge once tests are passing

**Comment by bleaker0x** [comment]
@trajan0x yeah that's my bad, apparently adding that index out of bounds check actually made the test fail... because it got the wrong error message. :P

**Comment by bleaker0x** [comment]
@trajan0x merge at your leisure.

**Pull request merged**